identity/src/lib.rs

```diff
@@ -87,14 +95,24 @@ pub mod account {
   //! Secure storage for Decentralized Identifiers
 
   pub use identity_account::account::*;
-  pub use identity_account::crypto::*;
   pub use identity_account::error::*;
   pub use identity_account::identity::*;
-  pub use identity_account::storage::*;
-  pub use identity_account::stronghold::*;
   pub use identity_account::types::*;
   pub use identity_account::updates::*;
-  pub use identity_account::utils::*;
+}
+
+#[cfg(feature = "account")]
+#[cfg_attr(docsrs, doc(cfg(feature = "account")))]
+pub mod account_core {
+  //! Storage Trait and Types definitions
+
+  pub use identity_account_core::crypto::*;
+  pub use identity_account_core::error::*;
+  pub use identity_account_core::identity::*;
+  pub use identity_account_core::storage::*;
+  pub use identity_account_core::stronghold::*;
+  pub use identity_account_core::types::*;
+  pub use identity_account_core::utils::*;
 }
```

Comment:
Discussion: I think we should re-export types from the `_core` crates under the non-core namespace (unless there are conflicts). That is, move all these `pub use identity_account_core` lines under `pub mod account`. Similarly for `identity_iota_core` and `identity_iota`.

All this does is simplify imports, which I think is desirable. The `Error` and `Result` instances are a problem, I'm not sure if we want to adopt a convention of prefixing the crate name onto those like in this PR where e.g. `IotaCoreError` is used instead of just `Error`? Not sure, need to decide and implement the convention everywhere.

Reply:
The convenience of condensed imports may not be worth any confusion, just to be clear.

Reply:
Overall I agree with re-exporting types from `_core` crates under the non-core namespace in `identity`, as consumers of the main `identity` are likely not interested in how it is organized internally. This same principle also applies to errors hence I would prefer to avoid variants/names like  `IotaCoreError` if possible. 

I am less sure about how much re-exports from sub(-sub) crates to unified modules in `identity` protect us against introducing breaking changes in the external API upon further sub-division however.  The types in our public API need to be interchangeable between minor versions, but cargo may not be able to "understand" that the types are actually still the same when their origin has changed from one sub crate to another. It might be possible to help express the compatibility of types between minor versions  by using something like [the semver trick](https://github.com/dtolnay/semver-trick).

Reply:
> I would prefer to avoid variants/names like IotaCoreError if possible.

The problem is that we cannot export two `Error` enums of the same name under `identity::iota` from `identity_iota` and `identity_iota_core` without changing the name of one or the other.

If we cannot avoid breaking changes by re-exporting types then I am somewhat less in favour of re-exporting them. The semver trick is fascinating but we should hopefully not plan to use it as a crutch.